components/supervisor/pkg/supervisor/supervisor.go

```diff
@@ -411,6 +413,17 @@ func Run(options ...RunOption) {
 		wg       sync.WaitGroup
 		shutdown = make(chan ShutdownReason, 1)
 	)
+	go func() {
+		allReady := waitForIde(ctx, ideReady, desktopIdeReady, ideReadyTimeoutDuration)
+		if !allReady {
+			msg := []byte(fmt.Sprintf("editor start timed out after %.0f minutes", ideReadyTimeoutDuration.Minutes()))
+			err := os.WriteFile("/dev/termination-log", msg, 0o644)
```

Comment:
is it a good place? Should not be there only one place for writing such log file?

Reply:
We need to wirte this file so that user knows what happend, otherwise it will show `supervisor run error with unexpected exit code: `

But yes, we can make it a function and change other callers

Reply:
I understand it, i just wonder whether there is another way to write to this file? i.e. trigger this code somehow https://github.com/gitpod-io/gitpod/blob/606963acdf12347bb3462eeb01e8a390c63f170a/components/supervisor/cmd/root.go#L56

although if we sure in this case it cannot race somehow with referenced code, maybe it is alright